mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -408,6 +451,119 @@ private[classification] class MultiClassSummarizer extends Serializable {
 }
 
 /**
+ * Abstraction for multinomial Logistic Regression Training results.
+ */
+sealed trait LogisticRegressionTrainingSummary extends Serializable {
+
+  /** objective function (scaled loss + regularization) at each iteration. */
+  val objectiveHistory: Array[Double]
+
+  /** Number of training iterations until termination */
+  val totalIterations: Int = objectiveHistory.length
```

Comment:
I think this is OK, but we should test this in a little Java suite.  Implementations within traits can cause Java API problems.